data/crac-creation/crac-creator-csa-profiles/src/main/java/com/farao_community/farao/data/crac_creation/creator/csa_profile/crac_creator/cnec/CsaProfileFlowCnecCreator.java

```diff
@@ -0,0 +1,390 @@
+    }
+
+    private void addFlowCnec(PropertyBag assessedElementPropertyBag) {
+        String assessedElementId = assessedElementPropertyBag.getId(CsaProfileConstants.REQUEST_ASSESSED_ELEMENT);
+        boolean isDataCheckOk = this.dataCheck(assessedElementId, assessedElementPropertyBag);
+
+        if (!isDataCheckOk) {
+            return;
+        }
+
+        PropertyBags assessedElementsWithContingencies = getAssessedElementsWithContingencies(assessedElementId, assessedElementPropertyBag);
+        if (assessedElementsWithContingencies.isEmpty()) {
+            return;
+        }
+
+        PropertyBags currentLimits = getCurrentLimits(assessedElementId, assessedElementPropertyBag);
+        if (currentLimits.isEmpty()) {
+            return;
+        }
+
+        String isCombinableWithContingencyStr = assessedElementPropertyBag.get(CsaProfileConstants.REQUEST_ASSESSED_ELEMENT_IS_COMBINABLE_WITH_CONTINGENCY);
+        Boolean isCombinableWithContingency = isCombinableWithContingencyStr != null && Boolean.parseBoolean(isCombinableWithContingencyStr);
+        Set<Contingency> combinableContingencies;
+        if (isCombinableWithContingency) {
+            combinableContingencies = cracCreationContext.getCrac().getContingencies();
+        } else {
+            combinableContingencies = new HashSet<>();
+        }
+
+        String nativeAssessedElementName = assessedElementPropertyBag.get(CsaProfileConstants.REQUEST_ASSESSED_ELEMENT_NAME);
+        String assessedSystemOperator = assessedElementPropertyBag.getId(CsaProfileConstants.REQUEST_ASSESSED_ELEMENT_OPERATOR);
+        String assessedElementName = CsaProfileCracUtils.getUniqueName(assessedSystemOperator, nativeAssessedElementName);
+
+        FlowCnecAdder flowCnecAdder = crac.newFlowCnec();
+        flowCnecAdder
+                .withMonitored(false)
+                .withOptimized(true)
+                .withReliabilityMargin(0);
+
+        for (PropertyBag currentLimit : currentLimits) {
+            if (!this.addCurrentLimit(assessedElementId, flowCnecAdder, currentLimit, isCombinableWithContingency)) {
+                return;
+            }
+        }
+
+        for (PropertyBag assessedElementWithContingencies : assessedElementsWithContingencies) {
+            combinableContingencies = this.checkLinkAssessedElementContingency(assessedElementId, assessedElementWithContingencies, combinableContingencies, isCombinableWithContingency);
+            if (combinableContingencies == null) {
+                return;
+            }
+        }
+
+        String inBaseCaseStr = assessedElementPropertyBag.get(CsaProfileConstants.REQUEST_ASSESSED_ELEMENT_IN_BASE_CASE);
+        Boolean inBaseCase = Boolean.parseBoolean(inBaseCaseStr);
+
+        for (Contingency contingency : combinableContingencies) {
+            String flowCnecName = assessedElementName + " - " + contingency.getName() + " - curative";
```

Comment:
It could also be auto / outage depending on the acceptableDuration

Reply:
corrected